Assets/Scripts/Control&Input/ItemSelectManager.cs

```diff
@@ -0,0 +1,300 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.InputSystem;
+using UnityEngine.EventSystems;
+using TMPro;
+using SecretName;
+
+public class ItemSelectManager : MonoBehaviour
+{
+    [SerializeField] 
+    private Item defaultBodyItem;
+    [SerializeField] 
+    private Item defaultBarrelItem;
+    [SerializeField] 
+    private Transform[] itemSpawnPoints;
+
+    [SerializeField]
+    private Canvas canvas;
+    
+    private List<GameObject> bodies = new List<GameObject>();
+    private List<GameObject> barrels = new List<GameObject>();
+    private List<GameObject> extensions = new List<GameObject>();
+    [SerializeField] 
+    private Timer timer;
+    private List<Item> bodyItems;
+    private List<Item> barrelItems;
+    private List<Item> extensionItems;
+    private InputManager inputManager;
+    [SerializeField] 
+    private Selectable defaultSelector;
+    private Vector2 moveInput;
+    private int bodyIndex;
+    private int barrelIndex;
+    private int extensionIndex;
+    [SerializeField]
+    private TMP_Text timerText;
+    private const float errorMarginInput = 0.1f;
+    private bool gamepadMoveReady = true;
+    public Transform cameraPosition;
+    [SerializeField]
+    private PlayerStatUI playerStatUI;
+    [SerializeField]
+    private TMP_Text secretName;
+
+    [SerializeField] 
+    private Image bodySelect;
+    [SerializeField] 
+    private Image barrelSelect;
+    [SerializeField] 
+    private Image extensionSelect; 
+    [SerializeField]
+    private Color selectedColor;
+    [SerializeField]
+    private Color defaultColor;
+    private int selectedIndex = 0;
```

Comment:
So this is for switching between types of parts, and should not have a generic name since we have indices for individual parts too. Perhaps call it `selectedTypeIndex` or `partTypeIndex`?

Reply:
I changed it to selectedTypeIndex.